WorkflowFunctionAdminBundle/Tests/Security/Authorization/Voter/WorkflowRightVoterTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace OpenOrchestra\WorkflowFunctionAdminBundle\Tests\Security\Authorization\Voter;
+
+use OpenOrchestra\WorkflowFunctionAdminBundle\Security\Authorization\Voter\WorkflowRightVoter;
+use Phake;
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+use OpenOrchestra\WorkflowFunction\Model\WorkflowRightInterface;
+use Doctrine\Common\Collections\ArrayCollection;
+
+/**
+ * Test WorkflowRightVoterTest
+ */
+class WorkflowRightVoterTest extends \PHPUnit_Framework_TestCase
+{
+    protected $workflowRightRepository;
+    protected $contentTypeRepository;
+    protected $contentType = 'fakeContentType';
+    protected $token;
+
+    /**
+     * @var WorkflowRightVoter
+     */
+    protected $voter;
+
+    /**
+     * Set up the test
+     */
+    public function setUp()
+    {
+        $contentType = Phake::mock('OpenOrchestra\ModelInterface\Model\ContentTypeInterface');
+        Phake::when($contentType)->getId()->thenReturn($this->contentType);
+
+        $this->contentTypeRepository = Phake::mock('OpenOrchestra\ModelInterface\Repository\ContentTypeRepositoryInterface');
+        Phake::when($this->contentTypeRepository)->findOneByContentTypeIdAndVersion(Phake::anyParameters())->thenReturn($contentType);
+
+        $this->workflowRightRepository = Phake::mock('OpenOrchestra\WorkflowFunction\Repository\WorkflowRightRepositoryInterface');
+
+        $user = Phake::mock('OpenOrchestra\UserBundle\Document\User');
+        Phake::when($user)->getId()->thenReturn('fakeUserId');
+
+        $this->token = Phake::mock('Symfony\Component\Security\Core\Authentication\Token\TokenInterface');
+        Phake::when($this->token)->getUser()->thenReturn($user);
+
+        $this->voter = new WorkflowRightVoter($this->workflowRightRepository, $this->contentTypeRepository);
+
+    }
+
+    /**
+     * Test instance
+     */
+    public function testInstance()
+    {
+        $this->assertInstanceOf('Symfony\Component\Security\Core\Authorization\Voter\VoterInterface', $this->voter);
+    }
+
+    /**
+     * @param string  $class
+     * @param boolean $isSupported
+     *
+     * @dataProvider provideClassName
+     */
+    public function testSupportsClass($class, $isSupported)
+    {
+        $this->assertEquals($isSupported, $this->voter->supportsClass($class));
+    }
+
+    /**
+     * @return array
+     */
+    public function provideClassName()
+    {
+        return array(
+            array(Phake::mock('stdClass'), false),
```

Comment:
I would add also the class ContentTypeInterface or StatusableInterface which should not be supported